sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPump.java

```diff
@@ -19,6 +23,9 @@ class PartitionPump implements AutoCloseable {
     private final EventHubConsumerAsyncClient client;
     private final Scheduler scheduler;
     private LastEnqueuedEventProperties lastEnqueuedEventProperties;
+    private final AtomicBoolean isDisposed = new AtomicBoolean();
+    // Period for scheduler to be disposed of gracefully. In normal times, it won't take this long time to dispose the scheduler.
+    private final Duration disposedTimeout = Duration.ofSeconds(1);
```

Comment:
This can probably be static. I am also curious how we know that 1 second is enough time? (Asking because the legacy client library waited up to 30 seconds before exiting.)

Reply:
1 second is enough for a update disposition request complete basically. the legacy client wait up to 30 seconds is for whole client, here 1 second is for a partition pump, if the client has more than 30 partitions, it may wait up more than 30 seconds in bad condition.
All in all, I can't give you a certain reason why I use 1 second for each partition pump, but 1 second really fix the problem in the issue.